### ISSUE TITLE
購入履歴画面と購入詳細画面のデータの保存

### ISSUE THREAD

**Issue by mine4080** [mine4080/lamp_practice#12]

**Comment by mine4080** [comment]
@primestyleus 

controllerの部分はfinish.phpで
if(purchase_carts($db, $carts) === false){
  set_error('商品が購入できませんでした。');
  redirect_to(CART_URL);
} else if  {
insert_purchace_history()
}
 で正常に商品が購入されれば、インサート文を実行する。

**Comment by primestyleus** [comment]
@mine4080 
ありがとうございます、いったんこちらで実装進めてください。

**Comment by mine4080** [comment]
@primestyleus 
こんにちは。

実装の続きで、if文に悩んでいます。
if(execute_query === false) {
} 
exexute_query が flase であれば、ロールバックを行えばいいのかと考えましたが、エラーが出て行き詰まっています。
ヒントを頂けますでしょうか。お願いします。

**Comment by primestyleus** [comment]
@mine4080 
はい、execute_queryのなかをかくにんいただくと、エラーがあったときにset_errorしているのがわかると思います。つまりエラーをカウントすればよいのですが、functions.php を見ると、それをやっている関数がありますので探してみましょう。
